pkg/api/types.go

```diff
@@ -189,6 +189,9 @@ type VolumeSource struct {
 	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"gcePersistentDisk"`
+	// AWSPersistentDisk represents a AWS EBS disk that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	AWSPersistentDisk *AWSPersistentDiskVolumeSource `json:"awsPersistentDisk"`
```

Comment:
I believe @bgrant0607's intention is to remove all these types of disk from VolumeSource and from the end user (pod authors and the like).  The user will make a request for a persistent volume (via PVClaim) and get something.  They won't know what and they shouldn't care.

The cluster admin, OTOH, deals with specific infrastructure.  He needs to create a specific kind of disk.  ```AWSElasticBlockStore``` maps cleanly to the AWS API type EbsBlockDevice.